lib/src/http_client_response.dart

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) 2021, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+part of '../src/http_client_request.dart';
+
+abstract class HttpClientResponse extends Stream<List<int>> {
```

Comment:
What is the need to wrap this class like this?

Reply:
We don't want to expose constructor of this class to the user.

Reply:
You can simply add a private constructor for that. 
```dart
class A {
  A._();
}
```
Now the user cannot create an object of this class. Since there is no public constructor.

Reply:
Okay nevermind this.